Implement nightly test workflow naming conventions

alisonshao opened the pull request:
## Summary
Restructures nightly test workflows with clear naming conventions and organizational structure.

## Changes

### 1. Parent Workflow
- Created `nightly-test-parent.yml` to orchestrate all child workflows
- Uses `workflow_call` pattern for better organization
- Includes comprehensive failure checking across all child workflows

### 2. Naming Convention
Implemented: `Nightly-test-{text/vlm}-{accuracy/perf}-X-gpu-{GPU model}`

**Text Model Workflows:**
- `nightly-test-text-accuracy-2-gpu-h100.yml`
- `nightly-test-text-perf-2-gpu-h100.yml`

**VLM Workflows:**
- `nightly-test-vlm-accuracy-2-gpu-h100.yml`
- `nightly-test-vlm-perf-2-gpu-h100.yml`

**GPU-Specific Workflows:**
- `nightly-test-general-1-gpu-h100.yml`
- `nightly-test-general-4-gpu-h100.yml`
- `nightly-test-general-8-gpu-h200.yml`
- `nightly-test-general-8-gpu-h20.yml`
- `nightly-test-perf-4-gpu-b200.yml`
- `nightly-test-perf-8-gpu-b200.yml`

**AMD Workflows:**
- `nightly-test-general-2-gpu-mi300.yml`
- `nightly-test-general-2-gpu-mi325.yml`

### 3. Enhanced Summary Tables
- Added `GPU_CONFIG` environment variable support
- Summary headers now include GPU configuration
- Example: `### deepseek-ai/DeepSeek-V3.1 (8-gpu-b200)`
- Modified `bench_one_batch_server.py` and `nightly_utils.py` to support this

### 4. Sequential Model Testing
- Each workflow tests models sequentially
- Easy to add more models to benchmarks

## Notes
- Old workflows (`nightly-test.yml`, `nightly-test-b200.yml`, `nightly-test-amd.yml`) remain in place for transition period
- All functionality from old workflows is preserved in new structure

alisonshao committed:
fix: create git tags directly instead of temporary branches

Changed release-fake-tag workflow to create tags directly,
eliminating the need to manually run version_branch_to_tag.sh

alisonshao committed:
Implement nightly test workflow naming conventions

Restructure nightly test workflows with clear naming and organization:

1. Parent workflow (nightly-test-parent.yml) orchestrates all child workflows
   using workflow_call pattern

2. Naming: Nightly-test-{text/vlm}-{accuracy/perf}-X-gpu-{GPU model}
   - Text: text-accuracy, text-perf (2-gpu-h100)
   - VLM: vlm-accuracy, vlm-perf (2-gpu-h100)
   - GPU-specific: 1/4/8-gpu for h100, h200, h20, b200
   - AMD: 2-gpu-mi300, mi325

3. Enhanced summary tables with GPU configuration
   - Added GPU_CONFIG environment variable
   - Headers show: "model-name (X-gpu-YYYY)"
   - Updated bench_one_batch_server.py and nightly_utils.py

Old workflows kept for transition period.

alisonshao committed:
revert unrelated change

alisonshao committed:
revert unrelated change

Add steps to create and push a fake tag in the workflow.

alisonshao committed:
revert unrelated change

Kangyan-Zhou commented on .github/workflows/nightly-test-parent.yml:
```diff
@@ -0,0 +1,116 @@
+  # Text model accuracy tests
+  nightly-test-text-accuracy-2-gpu-h100:
+    if: github.repository == 'sgl-project/sglang'
+    uses: ./.github/workflows/nightly-test-text-accuracy-2-gpu-h100.yml
```
Let's also create a folder called nightly and call all these workflows in the subfolder for better organizing them

alisonshao committed:
Organize nightly tests: Move to subfolder and create Nvidia/AMD/Intel parent workflows

- Moved all nightly test workflows to .github/workflows/nightly/ subfolder
- Created three parent workflows for better organization:
  * nightly-test-nvidia.yml - All Nvidia GPU tests (H100, H200, H20, B200)
  * nightly-test-amd.yml - All AMD GPU tests (MI300, MI325)
  * nightly-test-intel.yml - Placeholder for future Intel GPU tests
- Updated all workflow_call references to use nightly/ subfolder paths
- Replaced single parent workflow with vendor-specific parent workflows

Addresses PR feedback from @Kangyan-Zhou

alisonshao commented:
## Updates based on feedback

Addressed feedback from @Kangyan-Zhou:

### 1. Organized into three parent workflows
- **nightly-test-nvidia.yml** - All Nvidia GPU tests (H100, H200, H20, B200)
- **nightly-test-amd.yml** - All AMD GPU tests (MI300, MI325)
- **nightly-test-intel.yml** - Placeholder for future Intel GPU tests

### 2. Moved workflows to subfolder
- All nightly test workflows now in **.github/workflows/nightly/** subfolder
- Updated all `workflow_call` references to use nightly/ paths
- Better organization and cleaner workflow directory

### File Structure:
```
.github/workflows/
├── nightly-test-nvidia.yml    (parent)
├── nightly-test-amd.yml        (parent)
├── nightly-test-intel.yml      (parent)
└── nightly/
    ├── nightly-test-text-accuracy-2-gpu-h100.yml
    ├── nightly-test-text-perf-2-gpu-h100.yml
    ├── nightly-test-vlm-accuracy-2-gpu-h100.yml
    ├── nightly-test-vlm-perf-2-gpu-h100.yml
    ├── nightly-test-general-1-gpu-h100.yml
    ├── nightly-test-general-4-gpu-h100.yml
    ├── nightly-test-general-8-gpu-h200.yml
    ├── nightly-test-general-8-gpu-h20.yml
    ├── nightly-test-perf-4-gpu-b200.yml
    ├── nightly-test-perf-8-gpu-b200.yml
    ├── nightly-test-general-2-gpu-mi300.yml
    └── nightly-test-general-2-gpu-mi325.yml
```

alisonshao committed:
Merge remote-tracking branch 'origin/main' into feat/nightly-test-naming-conventions

alisonshao committed:
Distinguish model variants in summary headers

Fix issue where different model configurations (variants) showed identical
headers in summary tables. Now each variant is clearly labeled.

Changes:
- Update generate_markdown_report() to include run_name in header
- Pass variant as --run-name parameter to benchmark command
- Format: model-name (variant) [gpu-config]
- Example: 'deepseek-ai/DeepSeek-V3.1 (basic) [8-gpu-b200]'
           'deepseek-ai/DeepSeek-V3.1 (mtp) [8-gpu-b200]'

This fixes the confusion where models with different TP sizes, optimization
flags, or speculative decoding settings all showed the same header.

alisonshao commented:
Fixed model variant confusion in summaries. Different configurations now show distinct headers:
- `deepseek-ai/DeepSeek-V3.1 (basic) [8-gpu-b200]`
- `deepseek-ai/DeepSeek-V3.1 (mtp) [8-gpu-b200]`

Kangyan-Zhou reviewed (commented):
https://github.com/sgl-project/sglang/actions/runs/19347030632

alisonshao committed:
Fix workflow_call references: Move workflows to top level

GitHub Actions requires reusable workflows to be at the top level of
.github/workflows/ directory. Subdirectories are not supported for
workflow_call.

Changes:
- Moved all child workflows from nightly/ subdirectory back to top level
- Updated parent workflow references from nightly/nightly-test-* to nightly-test-*
- Removed empty nightly/ subdirectory

This fixes the workflow validation errors.

alisonshao committed:
Rename 1-gpu and 2-gpu workflows to use generic runner names

- Renamed nightly-test-*-h100.yml to nightly-test-*-runner.yml for 1-gpu and 2-gpu workflows
- Updated workflow names inside files to match new naming
- Changed GPU_CONFIG from '2-gpu-h100' to '2-gpu-runner'
- Updated parent workflow references to new filenames

alisonshao committed:
Trigger workflow test for Nvidia

alisonshao committed:
Fix test_deepseek_v32_nsabackend summary titles to distinguish variants

- Changed 'deepseek-v3' to 'deepseek-v3 flashmla' for flashmla backend
- Changed 'deepseek-v3' to 'deepseek-v3 fa3' for fa3 backend
- Changed 'deepseek-v3' to 'deepseek-v3 fp8kvcache' for fp8kvcache backend

alisonshao committed:
Display accuracy test results in table format

- Collect all test results from 3 variants in TEST_RESULTS list
- Generate markdown table showing variant, backends, kv cache, and accuracy
- Add GPU_CONFIG env var to 8-gpu workflows for proper headers
- Table format matches performance test summaries

alisonshao committed:
Inline child workflow jobs into parent Nvidia workflow

- Remove workflow_call pattern that created nested job layers
- Inline all 10 child workflow job definitions directly into parent
- Jobs now appear as flat list without /test suffix
- Preserve all configurations: runners, env vars, timeouts, steps

alisonshao committed:
Fix linting: Reformat dict appends with trailing commas

alisonshao committed:
Merge branch 'main' into feat/nightly-test-naming-conventions

alisonshao committed:
Remove continue-on-error: true from job level, keep --continue-on-error flag

- Remove continue-on-error: true from all nightly test workflow jobs
- Keep --continue-on-error flag in run_suite.py commands
- Jobs will now properly fail when tests fail
- check-all-jobs will correctly detect failures
- Follows implementation from PR #13096

alisonshao committed:
Merge branch 'main' into feat/nightly-test-naming-conventions

alisonshao committed:
Clean up nightly test workflow configuration

Removed temporary branches and paths for testing.

alisonshao committed:
Clean up nightly test workflow configuration

Removed temporary branches and paths from nightly test workflow.

alisonshao committed:
Delete unused child workflows and reorganize Nvidia test jobs

- Delete 10 unused child workflow files that would clutter Actions UI
- All jobs are now inline in nightly-test-nvidia.yml
- Reorganize jobs in logical order:
  1. General tests (1-gpu, 4-gpu, 8-gpu H200, 8-gpu H20)
  2. Text model tests (accuracy, perf)
  3. VLM tests (accuracy, perf)
  4. B200 performance tests (4-gpu, 8-gpu)
- Remove temporary branch trigger from push events
- Only 3 main workflows will appear: Nvidia, AMD, Intel

alisonshao committed:
Remove unused AMD child workflow files

The nightly-test-general-2-gpu-mi300.yml and nightly-test-general-2-gpu-mi325.yml
files are no longer used since nightly-test-amd.yml now uses a matrix strategy
to run tests on both MI300 and MI325 runners.
